service/notification/NotificationProcessor.php

```diff
@@ -156,10 +168,13 @@ public function processNotification($unprocessedNotification)
         // Process notifications based on it's event code
         switch ($unprocessedNotification['event_code']) {
             case AdyenNotification::AUTHORISATION:
-                // Notification success is 'true'
                 if ('true' === $unprocessedNotification['success']) {
-                    // Moves order to paid if order status is not paid already
-                    if ($this->isCurrentOrderStatusANonFinalStatus($order->getCurrentState())) {
+                    // If notification data does not match cart and order, set to PAYMENT_NEEDS_ATTENTION
+                    // Else if not in a final status, set to PAYMENT
+                    if (!$this->validateWithCartAndOrder($unprocessedNotification, $order)) {
```

Comment:
you can also do an early return since the notification is processed and the process is finished.
The elseif increasis the cognitive complexity and it makes it hard to understand the business logic
It's easier just to read like:

1. validate it - if it's not valid update order status and finish the process
2. update order status if ...
3. ...